include/seastar/core/metrics.hh

```diff
@@ -376,9 +376,9 @@ public:
     virtual ~metric_groups_def() = default;
     metric_groups_def(const metric_groups_def&) = delete;
     metric_groups_def(metric_groups_def&&) = default;
-    virtual metric_groups_def& add_metric(group_name_type name, const metric_definition& md) = 0;
-    virtual metric_groups_def& add_group(group_name_type name, const std::initializer_list<metric_definition>& l) = 0;
-    virtual metric_groups_def& add_group(group_name_type name, const std::vector<metric_definition>& l) = 0;
+    virtual metric_groups_def& add_metric(group_name_type name, const metric_definition& md, int handle) = 0;
+    virtual metric_groups_def& add_group(group_name_type name, const std::initializer_list<metric_definition>& l, int handle) = 0;
+    virtual metric_groups_def& add_group(group_name_type name, const std::vector<metric_definition>& l, int handle) = 0;
```

Comment:
Since `metric_groups_def` is a logical grouping, I wonder if it should be constructed with the handle, since each added metric should probably be registered with the same impl.

Reply:
I don't feel too strongly either, it just feels like the scope of the handle is more closely tied to the `metric_groups` than the metric/group.

Reply:
This took a lot longer than I expected. Testing this stuff is kind of painful. In any case, it's done. Feel free to reopen the conversation if needed.